erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -633,6 +633,29 @@ def make_item_gl_entries(self, gl_entries):
 								"project": item.project or self.project
 							}, account_currency, item=item))
 
+						# check if the exchange rate has changed
+						purchase_receipt_conversion_rate = frappe.db.get_value('Purchase Receipt', {'name': item.purchase_receipt}, ['conversion_rate'])
```

Comment:
Let's move this query out of the for loop and make a  function that returns a map will all the relevant details which we can use here

Reply:
Done.